src/Php82/Php82.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Php82;
+
+final class Php82
+{
+    // https://php.watch/versions/8.2/ini_parse_quantity#polyfill
+    public static function ini_parse_quantity(string $shorthand): int
+    {
+        $original_shorthand = $shorthand;
+        $multiplier = 1;
+        $sign = '';
+        $return_value = 0;
+
+        $shorthand = trim($shorthand);
+
+        // Return 0 for empty strings.
+        if ($shorthand === '') {
+            return 0;
+        }
+
+        // Accept + and - as the sign.
+        if ($shorthand[0] === '-' || $shorthand[0] === '+') {
+            if ($shorthand[0] === '-') {
+                $multiplier = -1;
+                $sign = '-';
+            }
+            $shorthand = substr($shorthand, 1);
+        }
+
+        // If there is no suffix, return the integer value with the sign.
+        if (preg_match('/^\d+$/', $shorthand, $matches)) {
+            return $multiplier * $matches[0];
+        }
+
+        // Return 0 with a warning if there are no leading digits
+        if (preg_match('/^\d/', $shorthand) === 0) {
+            trigger_error(
+                sprintf(
+                    'Invalid quantity "%s": no valid leading digits, interpreting as "0" for backwards compatibility',
+                    $original_shorthand
+                ),
+                E_USER_WARNING
+            );
+
+            return $return_value;
+        }
+
+        // Removing whitespace characters.
+        $shorthand = preg_replace('/\s/', '', $shorthand);
+
+        $suffix = strtoupper(substr($shorthand, -1));
+        switch ($suffix) {
+            case 'K':
+                $multiplier *= 1024;
+                break;
+            case 'M':
+                $multiplier *= 1024 * 1024;
+                break;
+            case 'G':
```

Comment:
This function is missing support for setting the base (using `o`, `x` or `b` characters). See: https://github.com/php/php-src/blob/master/Zend/zend_ini.c#L550

Please add them also in the test cases.

Reply:
@Ayesh I've tried to implement above on my fork - please have a look and maybe you can get some inspiration from there: https://github.com/IonBazan/polyfill/tree/feature/parse_ini_quantity